x-pack/solutions/security/plugins/security_solution/public/detection_engine/rule_creation_ui/pages/rule_editing/index.tsx

```diff
@@ -509,6 +525,15 @@ const EditRulePageComponent: FC<{ rule: RuleResponse }> = ({ rule }) => {
     [navigateToApp, ruleId]
   );
 
+  const updateCallToActionButton = useMemo(
```

Comment:
It would be great if we could move `updateCallToActionButton` and `isRuleUpgradeable` as well as `usePrebuiltRulesUpgrade` into a separate component, instead of defining them here.

Reply:
Refactored these files out